worker/uniter/runner/context/context_test.go

```diff
@@ -227,6 +228,47 @@ func (s *InterfaceSuite) TestConfigCaching(c *gc.C) {
 	c.Assert(settings, gc.DeepEquals, charm.Settings{"blog-title": "My Title"})
 }
 
+func (s *InterfaceSuite) TestGoalState(c *gc.C) {
+
+	mockUnitSince := func(inUnits application.UnitsGoalState) application.UnitsGoalState {
+		outUnits := application.UnitsGoalState{}
+		for name, gsStatus := range inUnits {
+			outUnits[name] = application.GoalStateStatus{
+				Status: gsStatus.Status,
+				Since:  application.GOAL_STATE_TIME_FORMAT,
+			}
+		}
+		return outUnits
+	}
+
+	gsStatus := application.GoalStateStatus{
+		Status: "waiting",
+		Since:  application.GOAL_STATE_TIME_FORMAT,
+	}
+	goalStateCheck := application.GoalState{
+		Units: application.UnitsGoalState{
+			"u/0": gsStatus,
+		},
+		Relations: map[string]application.UnitsGoalState{
+			"db": application.UnitsGoalState{
+				"u/0": gsStatus,
+			},
+		},
+	}
+
+	ctx := s.GetContext(c, -1, "")
+	goalState, err := ctx.GoalState()
+
+	// Mock status Since string
+	goalState.Units = mockUnitSince(goalState.Units)
```

Comment:
This is a test smell.

Reply:
What we do in status tests is to assert that the since times are not nil and then set the since times to nil before doing the test comparison wit the expected data